### DISCUSSION TITLE
Development roadmap of `devenv launch <application>`

### DISCUSSION THREAD

**Discussion by tai271828** [solvcon/devenv#23] [Ideas]
We have successfully moved forward for integration of SOLVCON and devenv https://github.com/solvcon/devenv/pull/22 . Inspired by the case study of SOLVCON ( https://github.com/solvcon/devenv/pull/22 ), we may want to *launch* an application / many applications in the future. Several perspectives pop up in my mind:

1. Why - do we really need such a feature/sub-command? Do we want to support different applications by default?
2. The name of the sub-command - it should hint the usage of the sub-command, and it won't be confused with existing sub-command names.

Regarding the item 1, I think the consensus is clear - Yes! Let me know if you have different thoughts.

Regarding the item 2, the first thing comes to my mind is: keeping the existing style as a verb may be a must. The current sub-command `add`, `use`, and `build` are all verbs, and they bring me very good user experience. For example, without a comprehensive README or manually, I can know how to use `devenv` by reading the sub-command names. I appreciate this kind of design, and hope it will be reserved and consistent.

**Comment by yungyuc** [comment]
I like the idea to have the app-specific command(s) in general and would like to hear more discussions about it.

**Comment by tai271828** [comment]
Tracking the issue with issue https://github.com/solvcon/devenv/issues/26

**Comment by tai271828** [comment]
We have a sciwork sprint today and I was working on this topic. I will summarize the discussion regarding `devenv launch` according to the sprint note later on. See the sprint agenda https://hackmd.io/UbOReM5CQcmtP5gL1zxqqw?view

**Reply by tai271828** [reply]
```
## Support devenv launch <application> #26
- `launch` is essentially `build`...
    - but may use `devenv` nestedly or recursively
    - possible implementation solution
        - thinking points
            - maintanence (code duplicate?)
            - readibility (naming `launch` vs `build`)
        - separate or consolidate/collect scripts?
```

**Reply by tai271828** [reply]
> We have a sciwork sprint today and I was working on this topic. I will summarize the discussion regarding `devenv launch` according to the sprint note later on. See the sprint agenda https://hackmd.io/UbOReM5CQcmtP5gL1zxqqw?view

Once https://github.com/solvcon/devenv/pull/37 is landed, it seems to me that `applications` is not orthogonal anymore from `devenv`. The namespace of `applications` is more like a role what `flavors` plays as. So, maybe I could/should move scripts under `applications` to `scripts`, and move `flavors/<flavor>/<application>` to `applications`.

**Comment by tai271828** [comment]
Renamed the subject from `support of devenv launch <application>` to `Development roadmap of  devenv launch <application>` because https://github.com/solvcon/devenv/issues/26 is closed and we would like to trace follow-up enhancement.

**Reply by tai271828** [reply]
> Renamed the subject from `support of devenv launch <application>` to `Development roadmap of devenv launch <application>` because #26 is closed and we would like to trace follow-up enhancement.

E.T.A. effort of `SOLVCON` and `devenv` integration is 1 man-week. It will take 3.5 month a.k.a. 14 weeks by committing 4 hours a week to this enhancement.

Original thread: https://github.com/solvcon/devenv/pull/37#discussion_r646087617
